File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<char,int>mpp;
        int l=0;
        
        int res=0;
        for(int r=0;r<s.size();r++){
            if(mpp.find(s[r])!=mpp.end()){
                l=max(mpp[s[r]]+1,l);
            }
            mpp[s[r]]=r;
            res=max(res,r-l+1);
        }
        return res;

        
    }
};

File: Data Structures & Algorithms/balanced-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public boolean isBalanced(TreeNode root) {
        int [] res=new int[1];
        dfs(root,res);
        if(res[0]==1){
            return false;
        }else{
            return true;
        }
        
    }
    public int dfs(TreeNode root, int[] res){
        if(root==null)return 0;
        int left=dfs(root.left,res);
        int right=dfs(root.right,res);
        if(Math.abs(right-left)>1){
            res[0]=1;
        }
        return 1+Math.max(left,right);
    }
}
